src/Umbraco.Core/Services/Implement/ContentTypeService.cs

```diff
@@ -14,20 +14,23 @@ namespace Umbraco.Core.Services.Implement
     /// </summary>
     internal class ContentTypeService : ContentTypeServiceBase<IContentTypeRepository, IContentType, IContentTypeService>, IContentTypeService
     {
-        public ContentTypeService(IScopeProvider provider, ILogger logger, IEventMessagesFactory eventMessagesFactory, IContentService contentService,
+        //Lazy to avoid circular dependency. It would have been better to refactor the delete logic into its own service
```

Comment:
@bergmania I think we should do this since this is not pretty. I also don't think that the `DeleteItemsOfTypes` really needs to belong under the ContentTypeService and that this delete method could be part of (or proxied/facade) through the IContentService and then we can avoid this circular dependency.

Reply:
@Shazwazza, The `DeleteItemsOfTypes `is overridden because its required by the `ContentTypeServiceBase`, where the `Delete` method is. If we move this to IContentService, we would end up having a depeneden from `ContentTypeServiceBase` to `IContentService`, and then the problem is almost the same.

Reply:
@bergmania ah right, so this might be another one of those refactor things. I think we could keep this Lazy thing going for now but make another task to refactor this so that we don't need to do that, like having a separate IContentDeleteService that can be injected

Reply:
@Shazwazza, I Totally agree, that is much cleaner :)